Fix MimirGoThreadsTooHigh runbook (#11845)
grafana/mimir: docs/sources/mimir/manage/mimir-runbooks/_index.md

- In Go, concurrency is handled via goroutines, which are lightweight threads managed by the Go runtime.
- Goroutines are multiplexed onto a small number of actual OS threads by the Go scheduler.
- Go threads are limited to 10K. When this limit is reached, the application panics with error like `runtime: program exceeds 10000-thread limit`.

> - If a goroutine makes a call to a blocking syscall (e.g. disk I/O, but not networking given it's implement using async I/O under the hood), the Go runtime will try to schedule other goroutines on other OS threads, starting new threads on-demand.

Reviewer: ```suggestion
- If a goroutine makes a call to a blocking syscall, for example, disk I/O, the Go runtime tries to schedule other goroutines on other OS threads. This process starts new threads on-demand.
```

Author: I would like to keep the networking part as a hint to not look at networking because the cause shouldn't be there. Can you rephrase it, keeping the networking part, please?

Reviewer: > Note that network syscalls use async I/O under the hood, so do not create this problem.

How about changing this to:

`Note that network syscalls use asynchronous I/O under the hood, and therefore, don't create this problem.`

Author: Sounds good. Can you edit the suggestion, to see the final corrected sentence, please?

- Idle go threads are never terminated ([issue](https://github.com/golang/go/issues/14592)), so once an application has a spike in the number of go threads, the process needs to be restarted to get back to a low number of threads.

How to **investigate**:

> - Check the process stack trace to find common patterns in where the goroutines blocked on syscall:

Reviewer: ```suggestion
- Check the process stack trace to find common patterns in where the goroutines are blocked on syscall:
```
